lib/brainstem/dsl/association.rb

```diff
@@ -37,12 +37,14 @@ def method_name
         end
       end
 
-      def run_on(model, context, helper_instance = Object.new)
+      def run_on(model, context, helper_instance = Object.new, lookup_options = {})
         if options[:lookup]
           run_on_with_lookup(model, context, helper_instance)
         elsif options[:dynamic]
           proc = options[:dynamic]
-          if proc.arity == 1
+          if proc.arity == -2
```

Comment:
Since the expected association proc format takes in `model` and `options = {}`, we start with arity 2, and with one of the params not required, makes it -2.

I'm not sure about the case at the bottom with no arity - as far as I know, our documentation requires that at least a model is passed in, resulting in arity 1?